include/stl2/detail/range/access.hpp

```diff
@@ -415,49 +449,42 @@ STL2_OPEN_NAMESPACE {
 		template <class R>
 		constexpr bool has_member = false;
 		template <class R>
-		requires
-			requires(const R& r) {
-				{ r.empty() } -> bool;
+		requires requires(const R& r) {
+				// { r.empty() } -> bool;
+				requires ConvertibleTo<__f<decltype(r.empty())>, bool>;
 			}
 		constexpr bool has_member<R> = true;
 
 		template <class R>
 		constexpr bool has_size = false;
 		template <class R>
-		requires
-			requires(const R& r) { __stl2::size(r); }
+		requires requires(const R& r) { __stl2::size(r); }
 		constexpr bool has_size<R> = true;
 
 		template <class R>
 		constexpr bool has_begin_end = false;
 		template <class R>
-		requires
-			requires(const R& r) {
-				__stl2::end(r);
-			}
+		requires requires { typename __end_t<R>; }
 		constexpr bool has_begin_end<R> = true;
 
 		struct fn {
 			// Prefer member
 			template <class R>
-			requires
-				has_member<R>
+			requires has_member<R>
 			constexpr bool operator()(const R& r) const
 			STL2_NOEXCEPT_RETURN(
 				static_cast<bool>(r.empty())
```

Comment:
Admittedly, this is not exactly `bool(r.empty())`, but I *think* the `static_cast` is equivalent here (perhaps only because `bool` is not a class type).